src/main/java/org/jenkinsci/plugins/orgfolder/github/MainLogic.java

```diff
@@ -0,0 +1,204 @@
+package org.jenkinsci.plugins.orgfolder.github;
+
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import hudson.BulkChange;
+import hudson.Extension;
+import hudson.model.AbstractItem;
+import hudson.model.AllView;
+import hudson.model.Descriptor;
+import hudson.model.Item;
+import hudson.model.ListView;
+import hudson.model.listeners.ItemListener;
+import hudson.util.DescribableList;
+import hudson.views.JobColumn;
+import hudson.views.ListViewColumn;
+import hudson.views.StatusColumn;
+import hudson.views.WeatherColumn;
+import jenkins.branch.Branch;
+import jenkins.branch.OrganizationFolder;
+import jenkins.model.Jenkins;
+import jenkins.scm.api.SCMSourceOwner;
+import jenkins.util.io.FileBoolean;
+import org.jenkinsci.plugins.github_branch_source.Connector;
+import org.jenkinsci.plugins.github_branch_source.GitHubSCMNavigator;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+import org.kohsuke.github.GHEvent;
+import org.kohsuke.github.GHHook;
+import org.kohsuke.github.GHOrganization;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GHUser;
+import org.kohsuke.github.GitHub;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.util.Arrays.*;
+
+/**
+ * Main logic of UI customization.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+@Extension
+public class MainLogic {
+    /**
+     * Applies UI customizations to {@link OrganizationFolder} for GitHub
+     */
+    public void applyOrg(OrganizationFolder of, GitHubSCMNavigator scm) throws IOException {
+        if (UPDATING.get().add(of)) {
+            BulkChange bc = new BulkChange(of);
+            try {
+                GitHub hub = connect(of, scm);
+                GHUser u = hub.getUser(scm.getRepoOwner());
+
+                of.setIcon(new GitHubOrgIcon());
+                of.getProperties().replace(new GitHubOrgProperty(u));
+                of.replaceAction(new GitHubLink("logo",u.getHtmlUrl()));
+                if (of.getDisplayNameOrNull()==null)
+                    of.setDisplayName(u.getName());
+                if (of.getView("Repositories")==null && of.getView("All") instanceof AllView) {
+                    // need to set the default view
+                    ListView lv = new ListView("Repositories");
+                    lv.getColumns().replaceBy(asList(
+                        new StatusColumn(),
```

Comment:
:bug: this can not be a `StatusColum()` because we are representing, in the first column, a repository (an icon).